Classes/Service/NodeRedirectService.php

```diff
@@ -159,10 +152,18 @@ protected function appendNodeAndChildrenDocumentsToPendingRedirects(NodeInterfac
             return;
         }
 
+        if ($node->isVisible() === false) {
+            return;
+        }
+
         if (!$this->hasNodeUriChanged($node, $targetWorkspace)) {
```

Comment:
This hasn't changed with this PR, but I run into this condition all the time, when I'm trying to test the redirects for deleted nodes.

Seems to me that a deleted node does not necessarily come with a detectable URI change.